iceoryx2-bb/container/src/byte_string.rs

```diff
@@ -242,22 +242,13 @@ impl<const CAPACITY: usize> FixedSizeByteString<CAPACITY> {
     pub fn from_bytes_truncated(bytes: &[u8]) -> Self {
         let mut new_self = Self::new();
         new_self.len = std::cmp::min(bytes.len(), CAPACITY);
-        unsafe {
-            std::ptr::copy(
-                bytes.as_ptr(),
-                new_self.data.as_ptr() as *mut u8,
-                bytes.len(),
-            )
-        };
+        for i in 0..new_self.len {
+            new_self.data[i].write(bytes[i]);
+        }
```

Comment:
Doesn't `new_self.data.write(bytes[0..new_self.len]);` work?

Reply:
Nope, since the array does not provide such a method. It is an array of MaybeUninits